compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/Scrollable.kt

```diff
@@ -278,33 +285,157 @@ private fun Modifier.pointerScrollable(
+}
+
+@Composable
+private fun Modifier.animatedMouseWheelScroll(
+    scrollLogic: State<ScrollingLogic>,
     mouseWheelScrollConfig: ScrollConfig,
-) = pointerInput(scrollingLogicState, mouseWheelScrollConfig) {
-    awaitPointerEventScope {
-        while (true) {
-            val event = awaitScrollEvent()
-            if (event.changes.fastAll { !it.isConsumed }) {
-                with(mouseWheelScrollConfig) {
-                    val scrollAmount = calculateMouseWheelScroll(event, size)
-                    with(scrollingLogicState.value) {
-                        val delta = scrollAmount.toFloat().reverseIfNeeded()
-                        val consumedDelta = scrollableState.dispatchRawDelta(delta)
-                        if (consumedDelta != 0f) {
-                            event.changes.fastForEach { it.consume() }
-                        }
+): Modifier {
+    var isAnimationRunning = false
+    val channel = remember { Channel<Float>(capacity = Channel.UNLIMITED) }
+    LaunchedEffect(scrollLogic) {
+        while (isActive) {
+            val event = channel.receive()
+            isAnimationRunning = true
+            scrollLogic.value.animatedDispatchScroll(event) {
+                // Sum delta from all pending events to avoid multiple animation restarts.
+                channel.sumOrNull()
+            }
+            isAnimationRunning = false
+        }
+    }
+    return mouseWheelInput(mouseWheelScrollConfig) {
+        with(scrollLogic.value) {
+            val delta = it.reverseIfNeeded().toFloat()
+            if (isAnimationRunning) {
+                channel.trySend(delta).isSuccess
+            } else {
+                // Try to apply small delta immediately to conditionally consume
+                // an input event and to avoid useless animation.
+                tryToScrollBySmallDelta(delta) {
+                    channel.trySend(it).isSuccess
+                }
+            }
+        }
+    }
+}
+
+private fun Channel<Float>.sumOrNull(): Float? {
+    val elements = untilNull { tryReceive().getOrNull() }.toList()
+    return if (elements.isEmpty()) null else elements.sum()
+}
+
+private fun <E> untilNull(builderAction: () -> E?) = sequence<E> {
+    do {
+        val element = builderAction()?.also {
+            yield(it)
+        }
+    } while (element != null)
+}
+
+private suspend fun ScrollingLogic.tryToScrollBySmallDelta(
+    delta: Float,
+    threshold: Float = 4f,
+    fallback: (Float) -> Boolean
+): Boolean {
+    var isConsumed = false
+    scrollableState.scroll {
+        isConsumed = if (abs(delta) > threshold) {
+            // Gather possibility to scroll by applying a piece of required delta.
+            val testDelta = if (delta > 0f) 1f else -1f
+            val consumedDelta = scrollBy(testDelta)
+            !consumedDelta.isAboutZero() && fallback(delta - testDelta)
+        } else {
+            val consumedDelta = scrollBy(delta)
+            !consumedDelta.isAboutZero()
```

Comment:
1. Good point
2. It looks right because it's count of pixels, so it will be rounded anyway and there is [existing place](https://github.com/JetBrains/compose-multiplatform-core/blob/fb5a7630ce40a89037fc70d717407a84017e4e75/compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/Scrollable.kt#L560-L561) with this approach.

Reply:
1. will you change this?
2. okay. But it is rounded in other place that don't have relationship with this code. rounding method can be changed there (toInt instead of round), and the code here won't behave as expected.

Reply:
1. Oh, I thought it about a different (but similar) line. I fixed it in cancel animation condition so it will make 1 frame difference there. But here this code is correct - we need to consume the event (return `true`) in case if any delta was consumed. I was like that before this change and I do not see the reason to change it.